src/formatting/types.rs

```diff
@@ -964,22 +964,21 @@ fn join_bounds_inner(
                 joiner
             };
 
-            let (trailing_str, extendable) = if i == 0 {
+            let (extendable, trailing_str) = if i == 0 {
                 let bound_str = item.rewrite(context, shape)?;
-                let bound_str_clone = bound_str.clone();
-                (bound_str, is_bound_extendable(&bound_str_clone, item))
+                (is_bound_extendable(&bound_str, item), bound_str)
```

Comment:
Neither related to the underlying bug nor necessary to fix it, just noticed while in here that we were unnecessarily cloning the formatted bound